sources/packages/api/src/auth/guards/authorized-parties.guard.ts

```diff
@@ -19,9 +26,52 @@ export class AuthorizedPartiesGuard implements CanActivate {
       return true;
     }
     const { user } = context.switchToHttp().getRequest();
+    const userToken = user as IUserToken;
 
-    return authorizedParties.some(
-      (authorizedParty) => authorizedParty === user.authorizedParty,
+    const hasAuthorizedParty = authorizedParties.some(
+      (authorizedParty) => authorizedParty === userToken.authorizedParty,
     );
+    if (!hasAuthorizedParty) {
+      throw new ForbiddenException(
+        "Client is not authorized under the expected authorized party(azp).",
+      );
+    }
+
+    const isAllowedIDP = this.isAllowedIDP(
+      userToken.authorizedParty,
+      userToken.IDP,
+    );
+    if (!isAllowedIDP) {
+      throw new ForbiddenException(
+        "Client is not authorized under the expected identity provider(IDP).",
+      );
+    }
+
+    return true;
+  }
+
+  /**
+   * Determines if the client is authorized through the expected IDP.
+   * @param authorizedParty authorized party type to be checked.
+   * @param idp identity provider used for authentication on Keycloak.
+   * @returns true if IDP is allowed, otherwise, false.
+   */
+  private isAllowedIDP(
+    authorizedParty: AuthorizedParties,
+    idp: IdentityProviders,
+  ): boolean {
+    switch (authorizedParty) {
```

Comment:
Why are we validating the IDP here? I also did not see the IDP validation in the user story.

Reply:
If multiple are allowed if just need to change the switch to accommodate it. I changed the switch to show how it looks like. Please take a look.
```typescript
switch (authorizedParty) {
      case AuthorizedParties.student:
      case AuthorizedParties.supportingUsers:
        return [IdentityProviders.BCSC, IdentityProviders.BCeID].includes(idp);
      case AuthorizedParties.institution:
        return idp === IdentityProviders.BCeID;
      case AuthorizedParties.aest:
        return idp === IdentityProviders.IDIR;
      case AuthorizedParties.formsFlowBPM:
        return true;
      default:
        return false;
    }
```